app/unit-tests/src/org/commcare/android/mocks/ModernHttpRequesterMock.java

```diff
@@ -60,17 +78,21 @@ public static void setRequestPayloads(String[] payloadReferences) {
     }
 
     @Override
-    protected HttpURLConnection setupConnection(URL builtUrl) throws IOException {
+    public Response<ResponseBody> makeRequest() throws IOException {
+        if(isAuthenticated && !new URL(url).getProtocol().contentEquals("https")){
```

Comment:
Is this necessary in the mock requester?

Reply:
This is added here to  to make the mockRequester behaves the same as the original one, since code that throws this exception moved to okHTTP interceptor which only gets called when we make an actual request.